client/src/retry.rs

```diff
@@ -0,0 +1,122 @@
+//! Turnkey Client to interact with the Turnkey API
+//! See <https://docs.turnkey.com>
+use std::time::Duration;
+
+#[derive(Debug, Clone)]
+pub struct RetryConfig {
+    /// Initial delay before the first retry
+    pub initial_delay: Duration,
+    /// Multiplier for `initial_delay`, yielding the next delay
+    pub multiplier: f64,
+    /// Maximum delay between retries to cap the time between retries.
+    pub max_delay: Duration,
+    /// Maximum number of retries.
+    pub max_retries: usize,
+}
+
+impl Default for RetryConfig {
+    fn default() -> Self {
+        RetryConfig {
+            initial_delay: Duration::from_millis(500),
+            multiplier: 2.0,
+            max_delay: Duration::from_secs(10),
```

Comment:
nit: the default config will never reach max_delay

Reply:
Good catch! I'll modify this.